MaxText/maxengine.py

```diff
@@ -0,0 +1,286 @@
+
```

Comment:
Can we move this file, "maxengine_server.py", and "maxengine_server.py" to its own sub-directory ( "jetstream server" or some better name) to have a clean separation of jetstream integration code?

Reply:
I'm not going to worry about this right now (I'm already the bottleneck for too much, things are simple enough and it isn't a one way door) but I'm happy for someone to take this on in the future.